### PR TITLE
Refactor bitesize4

### PR DESCRIPTION
**Pull request by t-young31**

### CODE REVIEW

File: tdms/include/field.h [hunk 1]
```diff
@@ -90,6 +105,11 @@ class SplitField : public Grid{
     double ***zx = nullptr;
     double ***zy = nullptr;
 
+    /**
+     * Has this split field been initialised from MATLAB? If it has then use a different free
```
Comment (dstansby) [inline tdms/include/field.h]: Is this sentence missing an ending?

File: tdms/include/field.h [hunk 2]
```diff
@@ -62,6 +63,13 @@ class Field : public Grid{
     xyz_arrays real;
     xyz_arrays imag;
 
+    /**
+     * Upper (u) and lower (l) indices in the x,y,z directions. e.g.
+     * il is the first non-pml cell in the i direction and iu the last in the corresponding split
+     * field grid
+     */
+    int il = 0, iu = 0, jl = 0, ju = 0, kl = 0, ku = 0;
```
Comment (dstansby) [inline tdms/include/field.h:71]: I presume these don't change during any one simuation?

File: tdms/src/iterator.cpp [hunk 3]
```diff
@@ -5971,56 +5793,54 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
...
+  
   auto interp_output_grid_labels = GridLabels();
 
   //fprintf(stderr,"Pos 15_m1\n");
   if (runmode == rm_complete && exphasorsvolume) {
     //now interpolate over the extracted phasors
     if (dimension == THREE) {
-      fprintf(stderr, "mxInterpolateFieldCentralE: %d %d %d \n", pind_iu - pind_il - 1,
-              pind_ju - pind_jl - 1, pind_ku - pind_kl - 1);
+      fprintf(stderr, "mxInterpolateFieldCentralE: %d %d %d \n", E.I_tot - 2,
```
Comment (dstansby) [inline tdms/src/iterator.cpp:5805]: Shouldn't these be `E.iu`, `E.il` etc. instead of `tot`?

Comment (dstansby) [reply tdms/src/iterator.cpp:5805]: And similar for all the changed lines below in this block

File: tdms/src/iterator.cpp [hunk 4]
```diff
@@ -6158,72 +5978,6 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
   if (exi_present) freeCastMatlab3DArray(exi, *Nt);
   if (eyi_present) freeCastMatlab3DArray(eyi, *Nt);
 
-  //fprintf(stderr,"Pos 18\n");
-  if (dimension == THREE) {
```
Comment (dstansby) [inline tdms/src/iterator.cpp:6162]: Is this casting being done elsewhere now, or is it redundant?

File: tdms/src/tensor_init.cpp [hunk 5]
```diff
@@ -0,0 +1,75 @@
+#include "string"
+#include "utils.h"
+#include "matlabio.h"
+#include "tensor_init.h"
+
+using namespace std;
+
+
+void init_grid_tensors(const mxArray *ptr, SplitField &E_s, SplitField &H_s, uint8_t*** &materials){
```
Comment (dstansby) [inline tdms/src/tensor_init.cpp:9]: Give this a quick docstring?

File: tdms/include/field.h (hunk 1 above)
Comment (t-young31) [reply tdms/include/field.h]: 🤦🏼 (I'm really bad at that) – thanks

File: tdms/include/field.h (hunk 2 above)
Comment (t-young31) [reply tdms/include/field.h:71]: nope

File: tdms/src/iterator.cpp (hunk 3 above)
Comment (t-young31) [reply tdms/src/iterator.cpp:5805]: I think this is okay:

```c++
 E.I_tot = E.iu - E.il + 1 = pind_iu - pind_il + 1
```
so
```c++
 E.I_tot - 2 == pind_iu - pind_il - 1
```

File: tdms/src/iterator.cpp (hunk 4 above)
Comment (t-young31) [reply tdms/src/iterator.cpp:6162]: the `freeX` functions are now being called in the class destructors, so should be okay 👍🏼

**Review by dstansby — APPROVED** [review]

**Pull request merged**